app/models/entities/account.rb

```diff
@@ -113,8 +113,12 @@ def discard!(attachment)
   # Class methods.
   #----------------------------------------------------------------------------
   def self.create_or_select_for(model, params)
+    account_by_name = Account.find_by(name: params[:name])
+
     if params[:id].present?
       account = Account.find(params[:id])
+    elsif account_by_name.present?
+      account = account_by_name
```

Comment:
Consider doing this as the following:

```
    elsif params[:name].present?
      account = Account.find_by(name: params[:name])
```

Reason being that you will execute a query when you run this line:
```
account_by_name = Account.find_by(name: params[:name])
```

Reply:
Good catch, after such time I don't know why I implemented it in this way.